src/lib/assets/search.utils.ts

```diff
@@ -32,7 +34,217 @@ export function searchAssetsWithNoMeta<T>(
     asset => {
       const slug = getSlug(asset);
       const [contract, tokenId] = fromAssetSlug(slug);
-      const metadata = isTezAsset(slug) ? TEZOS_METADATA : getMetadata(slug);
+      const metadata = getMetadata(slug);
+
+      return {
+        contract,
+        tokenId,
+        symbol: metadata?.symbol,
+        name: metadata?.name
+      };
+    }
+  );
+}
+
+export function searchTezosAssetsWithNoMeta<T>(
+  searchValue: string,
+  assets: T[],
+  getChainMetadata: (chainId: string, slug: string) => AssetMetadataBase | undefined,
+  getSlugWithChainId: (asset: T) => { chainId: string; assetSlug: string }
+) {
+  const trimmedSearchValue = searchValue.trim();
+
+  if (trimmedSearchValue.search(/^[A-Za-z0-9]+_\d*$/) === 0)
+    return assets.filter(asset => getSlugWithChainId(asset).assetSlug.startsWith(trimmedSearchValue));
+
+  return searchAndFilterItems(
+    assets,
+    searchValue,
+    Number.isInteger(Number(searchValue))
+      ? [
+          { name: 'tokenId', weight: 1 },
+          { name: 'symbol', weight: 0.75 },
+          { name: 'name', weight: 0.5 }
+        ]
+      : [
+          { name: 'symbol', weight: 1 },
+          { name: 'name', weight: 0.5 },
+          { name: 'contract', weight: 0.1 }
+        ],
```

Comment:
These configurations appear multiple times in this file. Please create constants for them